presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkDiskPageInput.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.common.Page;
+import com.facebook.presto.spark.classloader_interface.PrestoSparkDiskPage;
+import com.facebook.presto.spark.classloader_interface.PrestoSparkSerializedPage;
+import com.facebook.presto.spark.classloader_interface.PrestoSparkTaskOutput;
+import com.facebook.presto.spi.page.PagesSerde;
+import com.facebook.presto.spi.page.PagesSerdeUtil;
+import com.facebook.presto.spi.page.SerializedPage;
+import com.facebook.presto.spi.storage.TempDataOperationContext;
+import com.facebook.presto.spi.storage.TempStorage;
+import io.airlift.slice.InputStreamSliceInput;
+import org.apache.spark.SparkEnv;
+import org.apache.spark.storage.BlockManager;
+import org.apache.spark.storage.BroadcastBlockId;
+import scala.Option;
+import scala.reflect.ClassTag;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.spark.util.PrestoSparkUtils.computeChecksum;
+import static com.facebook.presto.spark.util.PrestoSparkUtils.toPrestoSparkSerializedPage;
+import static com.facebook.presto.spark.util.PrestoSparkUtils.toSerializedPage;
+import static java.util.Objects.requireNonNull;
+import static org.apache.spark.api.java.StorageLevels.MEMORY_AND_DISK;
+
+public class PrestoSparkDiskPageInput
+        implements PrestoSparkPageInput
+{
+    private static final Logger log = Logger.get(PrestoSparkDiskPageInput.class);
+
+    private final PagesSerde pagesSerde;
+    @GuardedBy("this")
+    private final List<Iterator<PrestoSparkSerializedPage>> serializedPageIterators;
+    @GuardedBy("this")
+    private int currentIteratorIndex;
+
+    public PrestoSparkDiskPageInput(PagesSerde pagesSerde,
```

Comment:
nit: wrap it. By Presto code style we either put all the parameters on the same line, or always put a parameter on  a new line